rpm2extents.c

```diff
@@ -239,181 +277,219 @@ static rpmRC process_package(FD_t fdi, FD_t digestori, FD_t validationi)
     uint32_t offset_ix = 0;
     size_t len;
     int next = 0;
+    struct rpmlead_s l;
+    rpmfiles files = NULL;
+    rpmfi fi = NULL;
+    char *msg = NULL;
 
     fdo = fdDup(STDOUT_FILENO);
 
-    if (rpmReadPackageRaw(fdi, &sigh, &h)) {
-	rpmlog(RPMLOG_ERR, _("Error reading package\n"));
-	exit(EXIT_FAILURE);
-    }
+    rc = rpmLeadReadAndReturn(fdi, &msg, &l);
+    if (rc != RPMRC_OK)
+	goto exit;
 
-    if (rpmLeadWrite(fdo, h))
-    {
-	rpmlog(RPMLOG_ERR, _("Unable to write package lead: %s\n"),
-		Fstrerror(fdo));
-	exit(EXIT_FAILURE);
-    }
+    /* Skip conversion if package is in deny list */
+    if (isInDenyList(l.name)) {
```

Comment:
probably good to write something to the logs using `rpmlog` when the package is in denylist so we know that conversion was skipped.

Reply:
Done here: https://github.com/chantra/rpm/compare/df1f8900cd6745b9aa8597d73cfa5028c680d48f..7976c921f60ec5d20c50c4c702ec5636b39210ba